hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -845,13 +844,14 @@ private List<DirectoryInfo> listAllPartitionsFromFilesystem(String initializatio
    * @param pendingDataInstants Files coming from pending instants are neglected
    * @return List consisting of {@code DirectoryInfo} for each partition found.
    */
-  private List<DirectoryInfo> listAllPartitionsFromMDT(String initializationTime, Set<String> pendingDataInstants) throws IOException {
+  private List<DirectoryInfo> listAllPartitionsFromMDT(StoragePath basePath, String initializationTime, Set<String> pendingDataInstants) throws IOException {
     List<String> allPartitionPaths = metadata.getAllPartitionPaths().stream()
         .map(partitionPath -> dataWriteConfig.getBasePath() + StoragePath.SEPARATOR_CHAR + partitionPath).collect(Collectors.toList());
     Map<String, List<StoragePathInfo>> partitionFileMap = metadata.getAllFilesInPartitions(allPartitionPaths);
     List<DirectoryInfo> dirinfoList = new ArrayList<>(partitionFileMap.size());
     for (Map.Entry<String, List<StoragePathInfo>> entry : partitionFileMap.entrySet()) {
-      dirinfoList.add(new DirectoryInfo(entry.getKey(), entry.getValue(), initializationTime, pendingDataInstants));
+      String relativeDirPath = FSUtils.getRelativePartitionPath(basePath, new StoragePath(entry.getKey()));
```

Comment:
What is exactly the effect from absolute path to relative for `DirectoryInfo`, like the bootstrap from MDT does not work correctly before.

Reply:
root issue is, within DirectoryInfo, we have a logic to deduce hoodie partitions by checking the presence of "_hoodie_partition_meta" file. This would work only when we do FS based listing. But for MDT based listing, we already have exact list of files for each partition. So, we do not need to deduce it. And infact the deduction logic results in no files returned from DirectoryInfo before this fix.

Reply:
specifically in this line .
Map<String, List<StoragePathInfo>> partitionFileMap = metadata.getAllFilesInPartitions(allPartitionPaths);

here the keys are representing entire path for partitions. i.e. /var/....tbl_path/partition/ 

But DirectoryInfo maintains only the relative path. i.e. "partition" . So, had to do this.